src/System.Text.Json/tests/Serialization/Value.ReadTests.cs

```diff
@@ -598,6 +618,817 @@ public static void ReadPrimitiveIReadOnlyListT()
             {
                 Assert.Equal(expected++, i);
             }
+
+            result = JsonSerializer.Parse<IReadOnlyList<int>>(Encoding.UTF8.GetBytes(@"[]"));
+            Assert.Equal(0, result.Count());
+        }
+
+        [Fact]
```

Comment:
Can you move these tests to their own file due to the large size.

Also as feasible, collapse similar tests \ add helpers to reduce the size.

Reply:
Done. I'm working on refactoring here https://github.com/dotnet/corefx/issues/37520.